tsl/src/bgw_policy/move_hyper_inval_api.c

```diff
@@ -0,0 +1,223 @@
+	namestrcpy(&check_name, POLICY_MOVE_HYPER_INVAL_CHECK_NAME);
+	namestrcpy(&check_schema, FUNCTIONS_SCHEMA_NAME);
+	namestrcpy(&owner, GetUserNameFromId(owner_id, false));
+
+	JsonbParseState *parse_state = NULL;
+
+	pushJsonbValue(&parse_state, WJB_BEGIN_OBJECT, NULL);
+	ts_jsonb_add_int32(parse_state, POLICY_CONFIG_KEY_HYPERTABLE_ID, ht->fd.id);
+
+	if (!include_tiered_data->isnull)
+		ts_jsonb_add_bool(parse_state,
+						  POLICY_CONFIG_KEY_INCLUDE_TIERED_DATA,
+						  include_tiered_data->value);
+
+	JsonbValue *result = pushJsonbValue(&parse_state, WJB_END_OBJECT, NULL);
+	Jsonb *config = JsonbValueToJsonb(result);
+
+	return ts_bgw_job_insert_relation(&application_name,
+									  schedule_interval,
+									  DEFAULT_MAX_RUNTIME,
+									  JOB_RETRY_UNLIMITED,
+									  schedule_interval,
+									  &proc_schema,
+									  &proc_name,
+									  &check_schema,
+									  &check_name,
+									  owner_id,
+									  true,
+									  fixed_schedule,
+									  ht->fd.id,
+									  config,
+									  initial_start,
+									  timezone);
+}
+
+Datum
+policy_move_hyper_inval_add(PG_FUNCTION_ARGS)
+{
+	Oid ht_oid = PG_GETARG_OID(0);
+	Interval *schedule_interval = PG_GETARG_INTERVAL_P(1);
+	const bool if_not_exists = PG_GETARG_BOOL(2);
+	TimestampTz initial_start = PG_ARGISNULL(3) ? DT_NOBEGIN : PG_GETARG_TIMESTAMPTZ(3);
+	const bool fixed_schedule = !PG_ARGISNULL(3);
+	text *const timezone = PG_ARGISNULL(4) ? NULL : PG_GETARG_TEXT_PP(4);
+	const NullableDatum include_tiered_data = {
+		.value = PG_GETARG_DATUM(5),
+		.isnull = PG_ARGISNULL(5),
+	};
+	char *valid_timezone = NULL;
+
+	ts_feature_flag_check(FEATURE_POLICY);
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+
+	if (fixed_schedule)
+	{
+		ts_bgw_job_validate_schedule_interval(schedule_interval);
+		if (TIMESTAMP_NOT_FINITE(initial_start))
+			initial_start = ts_timer_get_current_timestamp();
+	}
+
+	if (timezone != NULL)
+		valid_timezone = ts_bgw_job_validate_timezone(PG_GETARG_DATUM(6));
+
+	Cache *hcache;
+	Hypertable *ht = ts_hypertable_cache_get_cache_and_entry(ht_oid, CACHE_FLAG_NONE, &hcache);
+
+	/* Check that we have a continuous aggregate attached */
+	List *caggs = ts_continuous_aggs_find_by_raw_table_id(ht->fd.id);
+	if (list_length(caggs) == 0)
+		ereport(ERROR,
+				errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
+				errmsg("\"%s\" does not have a continuous aggregate attached",
+					   get_rel_name(ht_oid)));
+
+	Oid owner_id = ts_hypertable_permissions_check(ht_oid, GetUserId());
+	ts_bgw_job_validate_job_owner(owner_id);
+
+	int32 job_id = policy_move_hyper_inval_add_internal(ht,
+														if_not_exists,
+														schedule_interval,
+														owner_id,
+														initial_start,
+														fixed_schedule,
+														valid_timezone,
+														&include_tiered_data);
+
+	ts_cache_release(&hcache);
+
+	if (!TIMESTAMP_NOT_FINITE(initial_start))
+		ts_bgw_job_stat_upsert_next_start(job_id, initial_start);
```

Comment:
Another case that needs a test.

Reply:
Added a test that checks that the next_start is set correctly.